src/System.Text.Primitives/System/Text/Parsing/InvariantSigned.tt

```diff
@@ -0,0 +1,174 @@
+﻿<#@ template hostspecific="false" language="C#" #>
+<#@ output extension=".cs" #>
+<#@ import namespace="System.Collections.Generic" #>
+<#@ include file="PrimitiveParserHelpers.t4" #>
+<#
+
+List<TypeToParse> typesToParse = new List<TypeToParse>();
+
+typesToParse.Add(new TypeToParse {
+    PrimitiveName = "sbyte",
+    ClassName = "SByte",
+    ValueCalculationTypeName = "int"
+});
+
+typesToParse.Add(new TypeToParse {
+    PrimitiveName = "short",
+    ClassName = "Int16",
+    ValueCalculationTypeName = "int"
+});
+
+typesToParse.Add(new TypeToParse {
+    PrimitiveName = "int",
+    ClassName = "Int32",
+    ValueCalculationTypeName = "int"
+});
+
+typesToParse.Add(new TypeToParse {
+    PrimitiveName = "long",
+    ClassName = "Int64",
+    ValueCalculationTypeName = "long"
+});
+
+List<TryParseSignature> TryParseSignatures = new List<TryParseSignature>();
+
+TryParseSignatures.Add(new TryParseSignature {
+    BufferType = BufferType.ArrayPointer,
+    ConsumedParameterExists = false
+});
+
+TryParseSignatures.Add(new TryParseSignature {
+    BufferType = BufferType.ArrayPointer,
+    ConsumedParameterExists = true
+});
+
+TryParseSignatures.Add(new TryParseSignature {
+    BufferType = BufferType.ReadOnlySpan,
+    ConsumedParameterExists = false
+});
+
+TryParseSignatures.Add(new TryParseSignature {
+    BufferType = BufferType.ReadOnlySpan,
+    ConsumedParameterExists = true
+});
+
+List<Encoding> utfEncodings = new List<Encoding> { Encoding.Utf8, Encoding.Utf16 };
```

Comment:
Do you have to use real encoding classes? Could you use an array of strings new string [] { "Utf8", "Utf16" }?